src/utils/slpReader.ts

```diff
@@ -684,3 +685,49 @@ export function getGameEnd(slpFile: SlpFileType): GameEndType | null {
 
   return gameEndMessage as GameEndType;
 }
+
+export function extractFinalPostFrameUpdates(slpFile: SlpFileType): PostFrameUpdateType[] {
+  const { ref, rawDataPosition, rawDataLength, messageSizes } = slpFile;
+
+  // The following should exist on all replay versions
+  const postFramePayloadSize = messageSizes[Command.POST_FRAME_UPDATE];
+  const gameEndPayloadSize = messageSizes[Command.GAME_END];
+  const frameBookendPayloadSize = messageSizes[Command.FRAME_BOOKEND];
```

Comment:
Frame bookend message was only introduced in version 3.0.0 onwards, what happens if we try to extract the final post frame updates without this message size?

Reply:
In theory `frameBookendPayloadSize` should evaluate as undefined which will later cause the `frameBookendSize` variable to evaluate to zero. In short, it should work fine I think.